apps/meteor/client/views/omnichannel/directory/chats/contextualBar/AgentField.js

```diff
@@ -1,40 +1,40 @@
 import { useTranslation } from '@rocket.chat/ui-contexts';
-import React, { useMemo } from 'react';
+import React from 'react';
 
 import { UserStatus } from '../../../../../components/UserStatus';
 import UserAvatar from '../../../../../components/avatar/UserAvatar';
-import { AsyncStatePhase } from '../../../../../hooks/useAsyncState';
-import { useEndpointData } from '../../../../../hooks/useEndpointData';
 import AgentInfoDetails from '../../../components/AgentInfoDetails';
 import Field from '../../../components/Field';
 import Info from '../../../components/Info';
 import Label from '../../../components/Label';
 import { FormSkeleton } from '../../Skeleton';
+import { useUserInfo } from './hooks/useUserInfo';
 
 const AgentField = ({ agent, isSmall = false }) => {
 	const t = useTranslation();
 	const { username } = agent;
-	const { value, phase: state } = useEndpointData(
-		`/v1/users.info`,
-		useMemo(() => ({ username }), [username]),
-	);
 
-	if (state === AsyncStatePhase.LOADING) {
+	const { data, isLoading, isError } = useUserInfo(username);
+
+	if (!data || isLoading || isError) {
 		return <FormSkeleton />;
 	}
 
-	const {
-		user: { name, status },
-	} = value || { user: {} };
+	const { user } = data;
 
-	const displayName = name || username;
+	const displayName = user.name || username;
 
 	return (
 		<Field>
 			<Label>{t('Agent')}</Label>
 			<Info style={{ display: 'flex' }}>
```

Comment:
Just a heads up.  This `Info` is using a `Box` component underneath, so you could change this to a prop `display='flex'`.

Reply:
in fact!